power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -53,14 +198,62 @@ inline void get_component_sequence(MainModelState<ComponentContainer> const& sta
     return result;
 }
 
+// get sequence idx map of a certain batch scenario
+template <typename CompType, class ComponentContainer>
+std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> const& state,
+                                          ConstDataset const& update_data, Idx scenario_idx,
+                                          independence::UpdateCompProperties const& comp_independence = {}) {
+    auto const get_sequence = [&state, n_comp_elements = comp_independence.get_n_elements()](auto const& span) {
+        return get_component_sequence_by_it<CompType>(state, std::begin(span), std::end(span), n_comp_elements);
+    };
+    if (update_data.is_columnar(CompType::name)) {
+        auto const buffer_span =
+            update_data.get_columnar_buffer_span<meta_data::update_getter_s, CompType>(scenario_idx);
+        return get_sequence(buffer_span);
+    }
+    auto const buffer_span = update_data.get_buffer_span<meta_data::update_getter_s, CompType>(scenario_idx);
+    return get_sequence(buffer_span);
+}
+} // namespace detail
+
+template <class... ComponentTypes, class ComponentContainer>
+SequenceIdx<ComponentTypes...> get_all_sequence_idx_map(MainModelState<ComponentContainer> const& state,
+                                                        ConstDataset const& update_data, Idx scenario_idx,
+                                                        ComponentFlags<ComponentTypes...> const& components_to_store) {
+    return utils::run_functor_with_all_types_return_array<ComponentTypes...>(
+        [&update_data, &components_to_store, &state, scenario_idx]<typename CompType>() {
+            if (!std::get<index_of_component<CompType, ComponentTypes...>>(components_to_store)) {
+                return std::vector<Idx2D>{};
+            }
+            auto const n_components = state.components.template size<CompType>();
+            auto const independence = independence::check_component_independence<CompType>(update_data, n_components);
+            independence::validate_update_data_independence(independence);
+            return detail::get_component_sequence<CompType>(state, update_data, scenario_idx, independence);
+        });
+}
+// Get sequence idx map of an entire batch for fast caching of component sequences.
+// The sequence idx map of the batch is the same as that of the first scenario in the batch (assuming homogeneity)
+// This is the entry point for permanent updates.
+template <class... ComponentTypes, class ComponentContainer>
+SequenceIdx<ComponentTypes...> get_all_sequence_idx_map(MainModelState<ComponentContainer> const& state,
+                                                        ConstDataset const& update_data) {
+    constexpr ComponentFlags<ComponentTypes...> all_true = [] {
+        ComponentFlags<ComponentTypes...> result{};
+        std::ranges::fill(result, true);
+        return result;
+    }();
```

Comment:
Should this be dependent on the `update_data` rather than the whole picture?

Reply:
Because this overload of `get_all_sequence_idx_map` is only used for permanent updates, where only the first scenario is considered, hence it is always independent.

Reply:
Permanent updates are not necessarily updates of all components, right? What I meant to say is that yes it should be considered independent, but only for the components contained in the update data.

Reply:
You are right and I will take a look at it. However, is it worth to do it for specific components in this case though? Since it they are all going to be independent anyway, why search for the specific update ones?

Reply:
After looking at implementation, this would mean that we would need to iterate over each component type and then do a string look-up (via `update_data.find_component`) to determine which components from input are present in the update, and then only update those, which becomes unnecesarily expensive. 

So in this case, I will add a comment explaining why (it is easier and faster) this choice. Would that suffice?

Reply:
I don't think it's more expensive than the full list 'expansion'. My argument is that this way a vague communication between functions is introduced. Either we create a special state stating that it is `{'all', independent}` like in the very original implementation, or we always tailor the result. Think for example in the future when we debug this.

Reply:
the check is compile-time resolved. yes, you do do a loop over all component types, but also immediately check that the size is zero and early-out. checking for every component type whether it is in fact in the update data would be like implementing the same code twice. if there's any performance overhead, it should be negligible.

i prefer to keep the code as simple as possible. if changing the entry-point would make it more readable without significant performance overhead, go ahead, but otherwise, i see no reason to change this.

Reply:
That still does not address my question of returning something that does not truly reflect the state of the update data.

Reply:
soooo you're worried about the name `result`?
```suggestion
    constexpr ComponentFlags<ComponentTypes...> all_true = [] {
        ComponentFlags<ComponentTypes...> to_be_filled{};
        std::ranges::fill(to_be_filled, true);
        return to_be_filled;
    }();
```
maybe i'm missing which part of your question is about returning?

Reply:
good test case for a follow-up

Reply:
this conversation is still open

Reply:
Scoped out for a follow up PR